runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpUtils.kt

```diff
@@ -88,6 +80,19 @@ internal fun HttpRequest.toOkHttpRequest(
     return builder.build()
 }
 
+private fun Headers.toOkHttpHeaders(): OkHttpHeaders = OkHttpHeaders.Builder().also { okHeaders ->
```

Comment:
I forgot to say this but my only suggestion would be to have added this as an integration test that every engine that receives a gzip response leaves it alone and intact.

Reply:
Good idea! Added a test to the engine test suite and verified it failed before the fix and succeeds now.